### PR TITLE
Doc stakingv4

### PR DESCRIPTION
**Pull request by mariusmihaic**
#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [x] document new feature
- [ ] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/validators/staking-v4.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+:::
+
+![](/validators/stakingV4/current-staking.png)
+
+Nodes distribution is done in the following steps:
+
+1. For each shard, randomly shuffle out 80 eligible validators, resulting in 320(80 validators/4 shards) shuffled out
+   validators
+2. Select those 320 shuffled out validators to be randomly, but evenly, distributed at the **end of each shard's waiting
```
Comment (schimih) [inline docs/validators/staking-v4.md:85]:
```suggestion
2. Select those 320 shuffled-out validators to be randomly, but evenly, distributed at the **end of each shard's waiting
```

File: docs/validators/staking-v4.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+   list**
+3. For each shard, replace the previously shuffled out with 80 waiting validators from the **top of each shard's waiting
```
Comment (schimih) [inline docs/validators/staking-v4.md:87]:
```suggestion
3. For each shard, replace the previously shuffled-out with 80 waiting validators from the **top of each shard's waiting
```

File: docs/validators/staking-v4.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+**The first** and easy version it's a strict selection where we would sort all the nodes from the auction list based on
+their topUp and select the first ones.
+
+This is strict, as the VALIDATORS have to manage their number of nodes if their topUp is close to the border of
+selection. This is because the topUp is computed as TotalStake/NumberOfNodes - 2,5k eGLD. So if a validator registers
+more nodes, their topUp is smaller.
+Example:
+
+- 10 nodes, 50k eGLD = 2.5k topUp per node
```
Comment (schimih) [inline docs/validators/staking-v4.md:45]:
```suggestion
- 10 nodes, 50.000 eGLD = 2500 eGLD topUp per node
```

File: docs/validators/staking-v4.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+- 20 nodes, 50k eGLD = 0 topUp
```
Comment (schimih) [inline docs/validators/staking-v4.md:46]:
```suggestion
- 20 nodes, 50.000 eGLD = 0 eGLD topUp
```

File: docs/validators/staking-v4.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+## Staking v4. Phase 3.
+
+Starting with this epoch:
+
+- Maximum number of nodes in the network will be changed from 3200 to 2880 (3200 - 320), consisting of:
+    - a global number of 1600 active/eligible validators, split in 400 nodes/shard
+    - a global number of 1320 waiting validators to join the active list, split in 320 nodes/shard
+- All **shuffled out** nodes from eligible list will be sent to auction list in order to take part in the auction
+  selection. The more topUp an owner has, the higher the chances of having his auction nodes selected will be.
```
Comment (schimih) [inline docs/validators/staking-v4.md:147]:
```suggestion
- All **shuffled-out** nodes from the eligible list will be sent to the auction list to participate in the auction selection. The more topUp an owner has, the higher the chances of having his auction nodes selected will be.
```

File: docs/validators/staking-v4.md [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+- Based on the _soft auction selection_ (see next section), all **qualified** nodes from **auction** will be distributed
+  to waiting list (depending on the `available slots`). The other **unqualified** nodes will remain in auction and wait
```
Comment (schimih) [inline docs/validators/staking-v4.md:149]:
```suggestion
  to the waiting list (depending on the `available slots`). The other **unqualified** nodes will remain in the auction and wait
```

File: docs/validators/staking-v4.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+  to be selected in the next epoch, if possible. The number of available slots is based on the number of shuffled out
```
Comment (schimih) [inline docs/validators/staking-v4.md:150]:
```suggestion
  to be selected in the next epoch, if possible. The number of available slots is based on the number of shuffled-out
```

File: docs/validators/staking-v4.md [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+  nodes and other nodes leaving the network (e.g.: `unstake/jail`). It guarantees that the waiting list is filled and
+  the nodes configuration is maintained.
+- Distribution from `waiting` to `eligible` list will remain unchanged
```
Comment (schimih) [inline docs/validators/staking-v4.md:153]:
```suggestion
- Distribution from the `waiting` to the `eligible` list will remain unchanged
```

File: docs/validators/staking-v4.md [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+
+![](/validators/stakingV4/stakingV4-phase3.png)
+
+## Staking v4. Soft auction selection mechanism
+
+Nodes from the auction list will be selected to be distributed in the waiting list based on the **soft auction**
+mechanism.
+For each owner, based on his topUp, we compute how many validators he would be able to run by distributing his total
```
Comment (schimih) [inline docs/validators/staking-v4.md:161]:
```suggestion
For each owner, based on his topUp, we compute how many validators he could run by distributing his total
```

File: docs/validators/staking-v4.md [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+topUp per fewer nodes(considering we would not select all of his auction nodes, but only a part of them). This mechanism
+ensures that for each owner we select as many nodes as possible, based on the **minimum required topUp** to fill the
```
Comment (schimih) [inline docs/validators/staking-v4.md:163]:
```suggestion
ensures that for each owner, we choose as many nodes as possible, based on the **minimum required topUp** to fill the
```

File: docs/validators/staking-v4.md [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+`available slots`. This is a global selection, not per shard. We preselect the best global nodes at the end of epoch.
+
+Suppose we have the following auction list, and 3 available slots:
```
Comment (schimih) [inline docs/validators/staking-v4.md:166]:
```suggestion
Suppose we have the following auction list and 3 available slots:
```

File: docs/validators/staking-v4.md [hunk 12]
```diff
@@ -0,0 +1,297 @@
...
+| owner1 | 3                | 2                | 1                 | 3666         | 1222            | pubKey1                 |
+| owner2 | 3                | 1                | 2                 | 2555         | 851             | pubKey2, pubKey3        |
+| owner3 | 2                | 1                | 1                 | 2444         | 1222            | pubKey4                 |
+| owner4 | 4                | 1                | 3                 | 2666         | 666             | pubKey5, pubKe6, pubKe7 |
++--------+------------------+------------------+-------------------+--------------+-----------------+-------------------------+  
+```
+
+For the configuration above:
+
+- Minimum possible `topUp per node` = 666, considering `owner4` will have all of his **3 auction nodes selected**
```
Comment (schimih) [inline docs/validators/staking-v4.md:183]: Suggestion:
I would change numbers to get rid of the 666

File: docs/validators/staking-v4.md [hunk 13]
```diff
@@ -0,0 +1,297 @@
...
+- `owner4` has 3 auction nodes: `pubKey5`, `pubKey6`, `pubKey7` with a topUp per node = 666. By leaving two of his
+  nodes in auction(`pubKey6`, `pubKey7`), while only selecting one of them(`pubKey5`), his topUp per node would be
+  rebalanced to 1333 (2666/2) > 1216
+
+If we had increased the threshold above by one more step from `1216` to `1226`, we would only have two qualified
+nodes: `pubKey2` and `pubKey5`, which is not enough to fill all slots.
+
+:::note
+
+If an owner has multiple nodes in auction, but only a part of them are selected to be distributed in waiting list,
```
Comment (schimih) [inline docs/validators/staking-v4.md:226]:
```suggestion
If an owner has multiple nodes in auction, but only a part of them are selected to be distributed in the waiting list,
```

File: docs/validators/staking-v4.md [hunk 14]
```diff
@@ -0,0 +1,297 @@
...
+selection will be done by sorting the bls keys.
```
Comment (schimih) [inline docs/validators/staking-v4.md:227]:
```suggestion
selection will be made by sorting the bls keys.
```

File: docs/validators/staking-v4.md [hunk 15]
```diff
@@ -0,0 +1,297 @@
...
+
+:::
+
+:::note
+
+Minimum required topUp per node, as well as real time auction list will be available in explorer at all times so that
+owners can decide what's the best strategy to maximize the number of auction nodes that will be selected.
```
Comment (schimih) [inline docs/validators/staking-v4.md:234]:
```suggestion
The minimum required topUp per node, as well as real time auction list, will be public in explorer at all times so that owners can decide the best strategy to maximize the number of auction nodes selected.
```

File: docs/validators/staking-v4.md [hunk 16]
```diff
@@ -0,0 +1,297 @@
...
+
+:::
+
+Finally, we sort all validators based on qualified topUp per node and select them based on available slots. In case two
```
Comment (schimih) [inline docs/validators/staking-v4.md:238]:
```suggestion
Finally, we sort all validators based on qualified topUp per node and select them based on available slots. If two
```

File: docs/validators/staking-v4.md [hunk 17]
```diff
@@ -0,0 +1,297 @@
...
+or more validators have the same topUp(e.g.: `pubKey1` and `pubKey4`), the selection will be random, but deterministic.
```
Comment (schimih) [inline docs/validators/staking-v4.md:239]:
```suggestion
or more validators have the same topUp(e.g.: `pubKey1` and `pubKey4`), the selection will be random but deterministic.
```

File: docs/validators/staking-v4.md [hunk 18]
```diff
@@ -0,0 +1,297 @@
...
+Selection is done by a XOR between validators public keys and the current block's randomness. This ensures that
```
Comment (schimih) [inline docs/validators/staking-v4.md:240]:
```suggestion
Selection is made by an XOR between validators' public keys and the current block's randomness. This guarantees that
```

File: docs/validators/staking-v4.md [hunk 19]
```diff
@@ -0,0 +1,297 @@
...
+validators wouldn't be able to "mint" their bls keys in order to gain any selection advantage, since the randomness is
```
Comment (schimih) [inline docs/validators/staking-v4.md:241]:
```suggestion
validators will not be able to "mint" their bls keys to gain any selection advantage, since the randomness is
```

File: docs/validators/staking-v4.md [hunk 20]
```diff
@@ -0,0 +1,297 @@
...
+only known at the selection time.
```
Comment (schimih) [inline docs/validators/staking-v4.md:242]:
```suggestion
known only at the selection time.
```

File: docs/validators/staking-v4.md [hunk 21]
```diff
@@ -0,0 +1,297 @@
...
+ | owner3 | pubKey4        | 1222                     |
+ +--------+----------------+--------------------------+
+```
+
+Following our example above, we have two nodes with 1222 qualified top up per node:
+
+- `owner1` with 1 bls key = `pubKey1`
+- `owner3` with 1 bls key = `pubKey4`
+
+Suppose the result of the XOR operation between they bls keys and randomness is:
```
Comment (schimih) [inline docs/validators/staking-v4.md:261]:
```suggestion
Suppose the result of the XOR operation between their bls keys and randomness is:
```

File: docs/validators/staking-v4.md [hunk 22]
```diff
@@ -0,0 +1,297 @@
...
+
+- `XOR1` = `pubKey1` XOR `randomness` = `[143...]`
+- `XOR2` = `pubKey4` XOR `randomness` = `[131...]`
+
+Since `XOR1` > `XOR2`, `pubKey1` will to be selected, while `pubKey4` remains in the auction list.
```
Comment (schimih) [inline docs/validators/staking-v4.md:266]:
```suggestion
Since `XOR1` > `XOR2`, `pubKey1` will be selected, while `pubKey4` remains in the auction list.
```

File: docs/validators/staking-v4.md [hunk 23]
```diff
@@ -0,0 +1,297 @@
...
+
+## How much topUp should I have as a validator?
+
+## What happens if there are fewer nodes in auction than available slots?
+
+In this case all nodes will be selected, regardless of their topUP.
+
+## One of my nodes was sent to auction during stakingV4 phase2. Will I loose rewards?
+
+If one of your nodes is shuffled out into auction list during phase2 it will enter in competition with the other
```
Comment (schimih) [inline docs/validators/staking-v4.md:280]:
```suggestion
If one of your nodes is shuffled-out into auction list during the phase2, it will compete with the other
```

File: docs/validators/staking-v4.md [hunk 24]
```diff
@@ -0,0 +1,297 @@
...
+existing nodes. In case you have enough topUP, **nothing changes, no rewards will be lost**. For owners that
```
Comment (schimih) [inline docs/validators/staking-v4.md:281]:
```suggestion
existing nodes. If you have enough topUp, **nothing changes, and you won't lose any rewards**. THis change
```

File: docs/validators/staking-v4.md [hunk 25]
```diff
@@ -0,0 +1,297 @@
...
+contribute to the ecosystem and keep a nice topUp, this change will not have any negative impact. Of course, in case you
```
Comment (schimih) [inline docs/validators/staking-v4.md:282]:
```suggestion
will not negatively impact owners that contribute to the ecosystem and keep a nice topUp. Of course, if you have low topUp, or close to zero, your nodes might be unqualified and remain on the auction list.
```

File: docs/validators/staking-v4.md [hunk 26]
```diff
@@ -0,0 +1,297 @@
...
+have low topUp, or close to zero, your nodes might be unqualified and remain in auction list.
```
Comment (schimih) [inline docs/validators/staking-v4.md:283]:
```suggestion

```

File: docs/validators/staking-v4.md [hunk 27]
```diff
@@ -0,0 +1,297 @@
...
+
+## Why downsize the waiting list?
+
+Short answer: _to keep the APR unchanged_.
+
+Before stakingV4, in case a node was shuffled out and moved to waiting list, it was guaranteed that it would only be
+"idle"(not participating in consensus) for 5 epochs. During this time, the node would not gain any rewards.
+
+During stakingV4 phase2, no node from waiting is moved to active. If we were to keep the same configuration, a shuffled
+out node from this phase would have to wait 6 epochs until eligible(if selected from auction) and therefore decreasing
```
Comment (schimih) [inline docs/validators/staking-v4.md:293]:
```suggestion
out node from this phase would have to wait for 6 epochs until eligible(if selected from auction) and therefore decreasing
```

**Review by schimih — COMMENTED** [review]
Good text overall. I added just a few suggestions to bring some clarity...
Good job!

**Review by sasurobert — APPROVED** [review]

File: docs/validators/staking-v4.md [hunk 28]
```diff
@@ -0,0 +1,338 @@
...
+most nodes have a comfortable top-up on the base stake of 2.5k eGLD, some do not contribute to the network's security by
+adding more top-up.
+
+# **Problems with the Current Implementation**
+
+The issues with the current implementation include:
+
+- Limiting the number of nodes to 3200, creating an additional queue. New validators can join the network only if
+  someone leaves the system.
+- Concentration of power among large providers, hindering decentralization. MultiversX Foundation together with the top
```
Comment (sasurobert) [inline docs/validators/staking-v4.md:20]: I would rephrase it, say that the top X agencies control 33%.

File: docs/validators/staking-v4.md [hunk 29]
```diff
@@ -0,0 +1,338 @@
...
+  11 staking agencies control 33%.
+- Large providers strategically place nodes in the waiting list, taking advantage of their position when a slot becomes
```
Comment (sasurobert) [inline docs/validators/staking-v4.md:22]: I think this could be deleted.

File: docs/validators/staking-v4.md [hunk 30]
```diff
@@ -0,0 +1,338 @@
...
+
+1. Randomly shuffle out 80 eligible validators for each shard, resulting in 320 (80 validators per 4 shards)
+   shuffled-out validators.
+2. Select these 320 shuffled-out validators to be randomly but evenly distributed at the end of each shard's waiting
+   list.
+3. For each shard, replace the previously shuffled-out validators with 80 waiting validators from the top of each
+   shard's waiting list.
+
+In the current implementation, each node, regardless of its top-up, has equal chances of participating in the consensus.
+However, starting with staking phase 4, validators with higher staked top-up will have higher chances of participation.
```
Comment (sasurobert) [inline docs/validators/staking-v4.md:89]: actually those nodes with no topup, or small topup will have really small chance to enter into validation.

File: docs/validators/staking-v4.md [hunk 31]
```diff
@@ -0,0 +1,338 @@
...
+
+# **Staking V4**
+
+Staking phase 4 will unfold in three consecutive steps, each corresponding to a specific epoch.
+
+## Staking v4. Phase 1.
```
Comment (sasurobert) [inline docs/validators/staking-v4.md:95]: we named staking Phase 4. so I would change the "Phase" from this. It sounds band staking phase 4 phase 1. Just put step 1.

File: docs/validators/staking-v4.md [hunk 32]
```diff
@@ -0,0 +1,338 @@
...
+:::note
+
+If an owner has multiple nodes in the auction, but only a portion is selected for distribution in the waiting list, the
+selection will be based on sorting the BLS keys.
+
+:::
+
+:::note
+
+The minimum required topUp per node, along with the real-time auction list, will be accessible in the explorer at all
```
Comment (sasurobert) [inline docs/validators/staking-v4.md:233]: is accessible.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: docs/validators/staking-v4.md (hunk 28 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md:20]: Right, good point, rephrased it

File: docs/validators/staking-v4.md (hunk 29 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md:22]: Deleted it

File: docs/validators/staking-v4.md (hunk 30 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md:89]: Changed text to:

"Starting with staking phase 4, the probability of validators entering the validation process will be significantly influenced by the amount of their staked top-up. Validators with a higher staked top-up will have considerably greater chances of participation, while those with little or no top-up will find their chances of entering into validation markedly reduced."

File: docs/validators/staking-v4.md (hunk 31 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md:95]: Right, renamed everywhere to step

File: docs/validators/staking-v4.md (hunk 32 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md:233]: Refactored here and in another place as well

File: docs/validators/staking-v4.md [hunk 33]
```diff
@@ -0,0 +1,297 @@
...
+
+# **FAQ**
+
+## Will the APR decrease?
+
+## How much topUp should I have as a validator?
+
+## What happens if there are fewer nodes in auction than available slots?
+
+In this case all nodes will be selected, regardless of their topUP.
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply docs/validators/staking-v4.md]: Fixed

File: docs/validators/staking-v4.md (hunk 12 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md]: Fixed

File: docs/validators/staking-v4.md (hunk 3 above)
Comment (mariusmihaic) [reply docs/validators/staking-v4.md:41]: Fixed

File: docs/validators/staking-v4.md [hunk 34]
```diff
@@ -0,0 +1,297 @@
...
+- Big providers are getting bigger, it does not help with decentralization. 31% of validators are controlled by Elrond
+  Foundation. The top 11 staking agencies control 33%.
+- Big providers are putting up nodes in the waiting list - as they have little to lose and when somebody gets out, they
+  are the first to enter.
+
+One of the first things we should eliminate is the additional queue and use the topUp value per node in order to decide
+which are the best nodes and select those at the end of the epoch to become active. Doing that, we do not limit the
+entry of new validators as currently an old validator must leave in order for a new one to enter. The market will decide
+the actual price of the node, but it is not a hard auction, better say, it is a soft auction, where everybody paying the
+node price (2.5k eGLD) can register to the network, but the NODE becomes a validator only if it has enough TOPUP. The
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply docs/validators/staking-v4.md]: Fixed

File: docs/validators/staking-v4.md [hunk 35]
```diff
@@ -0,0 +1,297 @@
...
+title: Staking v4
+---
+
+# **Introduction**
+
+Staking and delegation should be developed as time goes on. No system has to stay the same. Our assumptions of how the
+market works and reacts may change over time, the same way it may change with users and markets.
+Right now we have around 400 validators (running one or multiple nodes). Some of them are staking providers, while
+others are individuals running validators directly. Although most of the nodes have a nice topUp on the top of the base
+stake of 2.5k eGLD, others don't contribute to the network's security by adding more topUp.
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply docs/validators/staking-v4.md]: Fixed 2.5k -> 2.500

**Review by sasurobert — APPROVED** [review]

**Review by schimih — APPROVED** [review]
